### ISSUE TITLE
Lack of reproducibility despite using a consistent seed

### ISSUE THREAD

**Issue by deanforce** [dmlc/xgboost#7287]
Hello,
We’ve been having some discussion concerning a lack of reproducibility despite using a consistent seed in xgboost when employing Bayesian optimization for the hyperparameters. After some digging, the issue seems to be in the xgboost package itself, but I’m still unclear as to what’s driving this. Could you please take a look? a sample syntax to reproduce the problem is available here: https://github.com/AnotherSamWilson/ParBayesianOptimization/issues/43 2

Thank you!

**Comment by trivialfis** [comment]
I think I got the result now.  It's caused by non-deterministic openmp reduction  Same cause as https://github.com/dmlc/xgboost/issues/5019 .
